File: Beecrowd/4-Estruturas_e_Bibliotecas/1023-Estiagem-10/1023.c
#include <stdio.h>
#include <math.h>
#define true 1
#define false 0
int max(int a, int b){return (a>b)? a:b;}
int main(){
    int cases, size, water, maxKey;
    double totalW, totalP;
    int c =1;
    int first = true;
    while(1){
        scanf("%d", &cases);
        if(cases==0)return 0;
        if(!first)printf("\n");
        first = false;
        int city[210] ={0};
        totalW = 0;
        totalP= 0;
        maxKey=0;
        for(int i=0;i<cases;i++){
            scanf("%d%d", &size, &water);
            totalP += size;
            totalW +=water;
            maxKey = max(maxKey, water/size);
            city[water/size] += size;  
        }
        printf("Cidade# %d:\n", c);
        for(int i =0;i<maxKey; i++){
            if(city[i] != 0)printf("%d-%d ", city[i], i);
        }
        printf("%d-%d\n", city[maxKey], maxKey);
        printf("Consumo medio: %.2lf m3.\n", floor((totalW/totalP)*100)/100);
        c++;
    }
}

File: Beecrowd/4-Estruturas_e_Bibliotecas/1023-Estiagem-10/1023.cs
using System;

class Program
{
    static void Main()
    {
        int cases, size, water, maxKey;
        double totalW, totalP;
        int c = 1;
        bool first = true;

        while (true)
        {
            cases = int.Parse(Console.ReadLine());
            if (cases == 0) return;
            if (!first) Console.WriteLine();
            first = false;

            int[] city = new int[210];
            totalW = 0;
            totalP = 0;
            maxKey = 0;

            for (int i = 0; i < cases; i++)
            {
                string[] input = Console.ReadLine().Split();
                size = int.Parse(input[0]);
                water = int.Parse(input[1]);
                totalP += size;
                totalW += water;
                maxKey = Math.Max(maxKey, water / size);
                city[water / size] += size;
            }

            Console.WriteLine($"Cidade# {c}:");
            for (int i = 0; i < maxKey; i++)
            {
                if (city[i] != 0) Console.Write($"{city[i]}-{i} ");
            }
            Console.WriteLine($"{city[maxKey]}-{maxKey}");
            Console.WriteLine($"Consumo medio: {Math.Floor((totalW / totalP) * 100) / 100:F2} m3.");
            c++;
        }
    }
}
